homeassistant/components/freebox/sensor.py

```diff
@@ -118,10 +122,40 @@ async def async_on_demand_update(self):
 
     async def async_added_to_hass(self):
         """Register state update callback."""
-        self._unsub_dispatcher = async_dispatcher_connect(
-            self.hass, self._router.signal_sensor_update, self.async_on_demand_update
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                self._router.signal_sensor_update,
+                self.async_on_demand_update,
+            )
         )
 
-    async def async_will_remove_from_hass(self):
-        """Clean up after entity before removal."""
-        self._unsub_dispatcher()
+
+class FreeboxCallSensor(FreeboxSensor):
+    """Representation of a Freebox call sensor."""
+
+    def __init__(
+        self, router: FreeboxRouter, sensor_type: str, sensor: Dict[str, any]
+    ) -> None:
+        """Initialize a Freebox call sensor."""
+        self._call_list_for_type = []
+        super().__init__(router, sensor_type, sensor)
+
+    def update(self) -> None:
+        """Update the Freebox call sensor."""
+        self._call_list_for_type = []
+        for call in self._router.call_list:
+            if not call["new"]:
+                continue
+            if call["type"] == self._sensor_type:
+                self._call_list_for_type.append(call)
+
+        self._state = len(self._call_list_for_type)
+
+    @property
+    def device_state_attributes(self) -> Dict[str, any]:
+        """Return device specific state attributes."""
+        return {
+            str(datetime.fromtimestamp(call["datetime"], TIMEZONE)): call["name"]
```

Comment:
We normally don't have dynamic keys in the state attributes. How will this be used?

Reply:
Not in automation (for me at least), it is to letting the user know who called and at what time.

Reply:
So that the information is directly here and he doesn't need to go to the Freebox app to know who called

Reply:
Maybe I can just remove the +0[X]:00 (date and time has been checked with the Freebox app, they are correct).
![image](https://user-images.githubusercontent.com/14821482/84960417-d7185080-b101-11ea-84ba-e55613472b7a.png)

Reply:
If these are real numbers that called you, you should maybe not show them here 😄

Reply:
Hahaha yeah sure !
But I actually don't have a phone at home (but I do have a phone number), so everyone that called here are people that I don't know: commercials or fake ... so I don't care :wink:

I used the `name` field here and not the `number` field of the call list, so when the number correspond to a contact, it should display its name instead of the number.

Reply:
Please use as many utils as possible from our dt.py util module.

Reply:
The datetime should be UTC time.

Reply:
The date time retrieved from the API is `Europe/Paris` timezoned.
If we use UTC the time displayed will have 1 ou 2h less (depending on summer time), right ?

How would you like this data to be displayed to the user ? (Freebox is only available in France)

Is the frontend handles date in attrs to display correctly formatted to the user ?
When I put date time as a key I got `Event is not JSON serializable`

---

Did some changes : (use French date format, as the router ISP is only available in France)

<img width="376" alt="Capture d’écran 2020-06-19 à 14 01 50" src="https://user-images.githubusercontent.com/14821482/85130825-f2847800-b235-11ea-8501-f872ac2e82c8.png">

Reply:
We can make it isoformatted with `.isoformat`.

All times in the state machine should be UTC time.